### PR TITLE
Add creation time to running task descriptor

### PR DESCRIPTION
**Pull request by RoboBaranik**

### CODE REVIEW

File: dicoogle/src/main/resources/webapp/js/components/indexer/TaskStatus.jsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ class TaskStatus extends React.Component {
         complete: PropTypes.bool,
         canceled: PropTypes.bool,
         taskProgress: PropTypes.number,
+        taskTimeCreated: PropTypes.number,
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/components/indexer/TaskStatus.jsx:18]: It should then become a string with the proposed change.

```suggestion
        taskTimeCreated: PropTypes.string,
```

File: dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js [hunk 2]
```diff
@@ -81,7 +81,10 @@ const IndexStatusView = createReactClass({
     if (this.state.data.tasks.length === 0) {
       items = <div>No tasks</div>;
     } else {
-      items = this.state.data.tasks.map(item => (
+      items = this.state.data.tasks.sort((item1, item2) => {
+        // Sorts tasks from latest to first
+        return item2.taskTimeCreated - item1.taskTimeCreated;
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js:86]:
```suggestion
        return new Date(item2.taskTimeCreated).getTime() - new Date(item1.taskTimeCreated).getTime();
```

**Review by Enet4 — CHANGES_REQUESTED** [review]
Thank you for this quick contribution! Please attend to the proposed changes before a more thorough review is given.

File: dicoogle/src/main/java/pt/ua/dicoogle/taskManager/RunningIndexTasks.java [hunk 3]
```diff
@@ -136,6 +137,7 @@ public JSONObject asJSON(Task<? extends Report> task) {
         entry.put("taskUid", task.getUid());
         entry.put("taskName", task.getName());
         entry.put("taskProgress", task.getProgress());
+        entry.put("taskTimeCreated", task.getTimeCreated().atZone(ZoneId.systemDefault()).toInstant().toEpochMilli());
```
[the remark this answers is not in the document]
Comment (RoboBaranik) [reply dicoogle/src/main/java/pt/ua/dicoogle/taskManager/RunningIndexTasks.java]: Changed to ISO String format.

File: dicoogle/src/main/resources/webapp/js/components/indexer/TaskStatus.jsx (hunk 1 above)
Comment (RoboBaranik) [reply dicoogle/src/main/resources/webapp/js/components/indexer/TaskStatus.jsx]: Fixed.

File: dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js (hunk 2 above)
Comment (RoboBaranik) [reply dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js]: Changed according to attribute format change.

**Review by Enet4 — APPROVED** [review]
Looks good to me! Redirecting to @bastiao for further testing, but I believe it's in good condition to be merged as is.

**Comment by bastiao** [comment]
I did not often missed the creating time of the task, but it's ok to add.
Thanks for your patch!

**Pull request merged**